collector/lib/CollectorConfig.cpp

```diff
@@ -438,17 +453,88 @@ void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
     YAML::Node yamlConfig = YAML::LoadFile(filePath);
     YamlConfigToConfig(yamlConfig);
   } catch (const YAML::BadFile& e) {
-    CLOG(FATAL) << "Failed to open the configuration file: " << filePath
-                << ". Error: " << e.what();
+    std::string errorMsg = "Failed to open the configuration file: " + filePath.string() + ". Error: " + e.what();
+    LogRuntimeConfigError(errorMsg);
   } catch (const YAML::ParserException& e) {
-    CLOG(FATAL) << "Failed to parse the configuration file: " << filePath
-                << ". Error: " << e.what();
+    std::string errorMsg = "Failed to parse the configuration file: " + filePath.string() + ". Error: " + e.what();
+    LogRuntimeConfigError(errorMsg);
   } catch (const YAML::Exception& e) {
-    CLOG(FATAL) << "An error occurred while loading the configuration file: " << filePath
-                << ". Error: " << e.what();
+    std::string errorMsg = "An error occurred while loading the configuration file: " + filePath.string() + ". Error: " + e.what();
+    LogRuntimeConfigError(errorMsg);
   } catch (const std::exception& e) {
-    CLOG(FATAL) << "An unexpected error occurred while trying to read: " << filePath << e.what();
+    std::string errorMsg = "An unexpected error occurred while trying to read: " + filePath.string() + e.what();
+    LogRuntimeConfigError(errorMsg);
+  }
+}
+
+void WaitForFileToExist(const std::filesystem::path& filePath) {
+  while (!std::filesystem::exists(filePath)) {
+    sleep(1);
+  }
+}
+
+void WaitForInotifyAddWatch(int& fd, int& wd, const std::filesystem::path& filePath) {
+  while (true) {
+    fd = inotify_init();
+    if (fd < 0) {
+      CLOG_THROTTLED(ERROR, std::chrono::seconds(30)) << "inotify_init failed";
+      continue;
+    }
+    int wd = inotify_add_watch(fd, filePath.c_str(), IN_MODIFY | IN_MOVE_SELF | IN_DELETE_SELF);
+    if (wd < 0) {
+      CLOG_THROTTLED(ERROR, std::chrono::seconds(30)) << "Failed to add inotify watch for " << filePath;
+      close(fd);
+    } else {
+      return;
+    }
+    sleep(1);
+  }
+}
```

Comment:
You only need to initialize inotify once, this method should look like this:
```suggestion
int WaitForInotifyAddWatch(int fd, const std::filesystem::path& filePath) {
  while (true) {
    int wd = inotify_add_watch(fd, filePath.c_str(), IN_MODIFY | IN_MOVE_SELF | IN_DELETE_SELF);
    if (wd < 0) {
      CLOG_THROTTLED(ERROR, std::chrono::seconds(30)) << "Failed to add inotify watch for " << filePath;
      close(fd);
    } else {
      return wd;
    }
    sleep(1);
  }
}
```